sql/src/test/java/io/crate/operation/aggregation/impl/PercentileAggregationTest.java

```diff
@@ -0,0 +1,186 @@
+    @Test
+    public void testReturnType() throws Exception {
+        FunctionIdent fi = new FunctionIdent("percentile", ImmutableList.<DataType>of(DataTypes.LONG, DataTypes.DOUBLE));
+        assertEquals(DataTypes.DOUBLE, functions.get(fi).info().returnType());
+    }
+
+    @Test
+    public void testInteger() throws Exception {
+        Object[][] result = executeAggregation(DataTypes.INTEGER, new Object[][]{
+            {1, 0.5},
+            {2, 0.5},
+            {3, 0.5},
+            {4, 0.5},
+            {5, 0.5},
+            {6, 0.5},
+            {7, 0.5},
+            {8, 0.5},
+            {9, 0.5},
+            {10, 0.5}
+        });
+
+        assertEquals(5.5, result[0][0]);
+    }
+
+    @Test
+    public void testDouble() throws Exception {
+        Object[][] result = executeAggregation(DataTypes.DOUBLE, new Object[][]{
+            {1.0, 0.5},
+            {2.0, 0.5},
+            {3.0, 0.5},
+            {4.0, 0.5},
+            {5.0, 0.5},
+            {6.0, 0.5},
+            {7.0, 0.5},
+            {8.0, 0.5},
+            {9.0, 0.5},
+            {10.0, 0.5}
+        });
+
+        assertEquals(5.5, result[0][0]);
+    }
+
+    @Test
+    public void testLong() throws Exception {
+        Object[][] result = executeAggregation(DataTypes.LONG, new Object[][]{
+            {1L, 0.5},
+            {2L, 0.5},
+            {3L, 0.5},
+            {4L, 0.5},
+            {5L, 0.5},
+            {6L, 0.5},
+            {7L, 0.5},
+            {8L, 0.5},
+            {9L, 0.5},
+            {10L, 0.5}
+        });
+
+        assertEquals(5.5, result[0][0]);
+    }
+
+    @Test
+    public void testShort() throws Exception {
+        Object[][] result = executeAggregation(DataTypes.SHORT, new Object[][]{
+            {(short) 1, 0.5},
+            {(short) 2, 0.5},
+            {(short) 3, 0.5},
+            {(short) 4, 0.5},
+            {(short) 5, 0.5},
+            {(short) 6, 0.5},
+            {(short) 7, 0.5},
+            {(short) 8, 0.5},
+            {(short) 9, 0.5},
+            {(short) 10, 0.5}
+        });
+
+        assertEquals(5.5, result[0][0]);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testNegativePercentile() throws Exception {
+        Object[][] result = executeAggregation(DataTypes.INTEGER, new Object[][]{
+            {1, -0.5},
+            {10, -0.5}
+        });
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testTooLargePercentile() throws Exception {
+        Object[][] result = executeAggregation(DataTypes.INTEGER, new Object[][]{
+            {1, 3.0},
+            {10, 3.0}
+        });
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testUnsupportedType() throws Exception {
+        Object[][] result = executeAggregation(DataTypes.STRING, new Object[][]{
+            {"Akira", 0.5},
+            {"Tetsuo", 0.5}
+        });
+    }
+
+    @Test
+    public void testStreaming() throws Exception {
```

Comment:
Would this new test class be best placed in the same folder as AggregationPercentileTest?